File: examples/key/new_managed_key/new_managed_key.go
package main

import (
	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
)

func main() {
	keyClient := client.NewKeyClient()
	protection := key.KEY_PROTECTION_SOFTWARE

	// Generate a EcP256k key
	keyType := key.EcP256k
	_, _ = keyClient.NewManagedKey(key.ManagedKeyParams{
		Protection: protection,
		KeyType:    keyType,
	})

	// Generate a Rsa2048 key
	keyType = key.Rsa2048
	_, _ = keyClient.NewManagedKey(key.ManagedKeyParams{
		Protection: protection,
		KeyType:    keyType,
	})

	// Generate a Rsa3072 key
	keyType = key.Rsa3072
	_, _ = keyClient.NewManagedKey(key.ManagedKeyParams{
		Protection: protection,
		KeyType:    keyType,
	})
	// Generate a Rsa4096 key
	keyType = key.Rsa4096
	_, _ = keyClient.NewManagedKey(key.ManagedKeyParams{
		Protection: protection,
		KeyType:    keyType,
	})

	// Generate a Aes128 key
	keyType = key.Aes128
	_, _ = keyClient.NewManagedKey(key.ManagedKeyParams{
		Protection: protection,
		KeyType:    keyType,
	})

	// Generate a Aes256 key
	keyType = key.Aes256
	_, _ = keyClient.NewManagedKey(key.ManagedKeyParams{
		Protection: protection,
		KeyType:    keyType,
	})

	// Generate a BJJ key
	keyType = key.Bjj
	_, _ = keyClient.NewManagedKey(key.ManagedKeyParams{
		Protection: protection,
		KeyType:    keyType,
	})
}

File: examples/key/new_managed_key/new_managed_key.ts
import {
  KeyClient,
  KeyProtectionLevel,
  KeyType,
  ManagedKeyParams,
} from '@bloock/sdk';

const keyClient = new KeyClient();
const keyProtection = KeyProtectionLevel.SOFTWARE;

// Generate a EcP256k key
let keyType = KeyType.EcP256k;
let _key = await keyClient.newManagedKey(
  new ManagedKeyParams(keyProtection, keyType)
);

// Generate a Rsa2048 key
keyType = KeyType.Rsa2048;
_key = await keyClient.newManagedKey(
  new ManagedKeyParams(keyProtection, keyType)
);

// Generate a Rsa3072 key
keyType = KeyType.Rsa3072;
_key = await keyClient.newManagedKey(
  new ManagedKeyParams(keyProtection, keyType)
);

// Generate a Rsa4096 key
keyType = KeyType.Rsa4096;
_key = await keyClient.newManagedKey(
  new ManagedKeyParams(keyProtection, keyType)
);

// Generate a Aes128 key
keyType = KeyType.Aes128;
_key = await keyClient.newManagedKey(
  new ManagedKeyParams(keyProtection, keyType)
);

// Generate a Aes256 key
keyType = KeyType.Aes256;
_key = await keyClient.newManagedKey(
  new ManagedKeyParams(keyProtection, keyType)
);

// Generate a Bjj key
keyType = KeyType.Bjj;
_key = await keyClient.newManagedKey(
  new ManagedKeyParams(keyProtection, keyType)
);
